Assets/Scripts/Puzzle/ItemPlace.cs

```diff
@@ -0,0 +1,35 @@
+using UnityEngine;
+using System;
+public class ItemPlace : MonoBehaviour
+{
+    [SerializeField] string _itemName;
+    [SerializeField] private InventoryUI _inventoryUI;
```

Comment:
referencing inventory ui for all the item places seems bad to me, how about making the inventory ui a singleton?